### PR TITLE
[SofaKernel] several small fix

### PR DESCRIPTION
**Pull request by epernod**
Several changes and fix from my master branch:
- Fix: Duplicate declaration of Data in MechanicalObject
- Fix: the graph display of topologyData.
- Replace serr by msg_error and put some methods and parameter from private to protected in SofaMeshCollision classes
- Add: missing generic methods in MeshTopology
______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaBaseTopology/MeshTopology.cpp [hunk 1]
```diff
@@ -1589,6 +1589,19 @@ const MeshTopology::HexahedraAroundQuad& MeshTopology::getHexahedraAroundQuad(Qu
 
 
 
+const vector< MeshTopology::EdgesAroundVertex >& MeshTopology::getEdgesAroundVertexArray()
+{
+    if (m_edgesAroundVertex.empty())	// this method should only be called when the array exists.
+    {
+#ifndef NDEBUG
+        sout << "Warning. [MeshTopology::getEdgesAroundVertexArray] EdgesAroundVertex array is empty." << sendl;
+#endif
```
Comment (guparan) [inline SofaKernel/modules/SofaBaseTopology/MeshTopology.cpp]: `msg_warning() << "getEdgesAroundVertexArray: EdgesAroundVertex array is empty.";`

What is the purpose of NDEBUG?

File: SofaKernel/modules/SofaMeshCollision/PointLocalMinDistanceFilter.cpp [hunk 2]
```diff
@@ -244,7 +244,7 @@ void PointLocalMinDistanceFilter::handleTopologyChange()
 {
     if(this->isRigid())
     {
-        serr<<"WARNING: filters optimization needed for topological change on rigid collision model"<<sendl;
+        msg_error() << "WARNING: filters optimization needed for topological change on rigid collision model";
```
Comment (guparan) [inline SofaKernel/modules/SofaMeshCollision/PointLocalMinDistanceFilter.cpp]: Error or warning?

File: SofaKernel/modules/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp [hunk 3]
```diff
@@ -190,7 +190,7 @@ void TriangleLocalMinDistanceFilter::handleTopologyChange()
 {
     if(this->isRigid())
     {
-        serr<<"WARNING: filters optimization needed for topological change on rigid collision model"<<sendl;
+        msg_error() << "WARNING: filters optimization needed for topological change on rigid collision model";
```
Comment (guparan) [inline SofaKernel/modules/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp]: Error or warning?

File: SofaKernel/modules/SofaBaseMechanics/MechanicalObject.h [hunk 4]
```diff
@@ -96,19 +96,11 @@ class MechanicalObject : public sofa::core::behavior::MechanicalState<DataTypes>
 public:
     virtual void parse ( core::objectmodel::BaseObjectDescription* arg ) override;
 
-#ifdef SOFA_HAVE_NEW_TOPOLOGYCHANGES
+#ifdef SOFA_HAVE_NEW_TOPOLOGYCHANGES
```
Comment (guparan) [inline SofaKernel/modules/SofaBaseMechanics/MechanicalObject.h:99]: Looks like you slipped on the space bar here :wink:

File: SofaKernel/modules/SofaMeshCollision/PointLocalMinDistanceFilter.cpp (hunk 2 above)
Comment (epernod) [reply SofaKernel/modules/SofaMeshCollision/PointLocalMinDistanceFilter.cpp]: same same...

File: SofaKernel/modules/SofaBaseMechanics/MechanicalObject.h (hunk 4 above)
Comment (epernod) [reply SofaKernel/modules/SofaBaseMechanics/MechanicalObject.h:99]: is there an easy way to remove that without doing a commit only for that?

Comment (guparan) [reply SofaKernel/modules/SofaBaseMechanics/MechanicalObject.h:99]: Unfortunately no.

**Pull request merged**